chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -109,7 +109,8 @@ impl EthereumAdapter {
 
         // Check if the provider supports `getBlockReceipts` method.
         let supports_block_receipts =
-            Self::check_block_receipt_support(web3.clone(), &provider, &logger).await;
+            Self::check_block_receipt_support(web3.clone(), &provider, supports_eip_1898, &logger)
```

Comment:
worth adding some tests?

Reply:
@mangas just added the tests, could you review?